drivers/ieee802154/ieee802154_nrf5.c

```diff
@@ -650,6 +688,69 @@ static void nrf5_iface_init(struct net_if *iface)
 	ieee802154_init(iface);
 }
 
+#if defined(CONFIG_NRF_802154_ENCRYPTION)
+static void nrf5_config_mac_keys(struct ieee802154_key *mac_keys)
+{
+	nrf_802154_security_error_t err;
+	nrf_802154_key_t key;
+	uint8_t key_id_to_remove;
+
+	__ASSERT(mac_keys, "Invalid argument.");
+
+	/* Remove old invalid key assuming that its index is first_valid_key_id - 1.
+	 * TODO: This is Thread specific assumption, need to be changed when RD will provided
+	 * API for removing all keys or handling this internally.
+	 */
+	key_id_to_remove = (*mac_keys).key_index == 1 ? 0x80 : (*mac_keys).key_index - 1;
```

Comment:
Throughout this function there is `(*mac_keys).sth`. Customary in C we use `mac_keys->sth`. It would be easier to follow the code if `->` was used instead.